### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by AndriiYelieva**
[DEMO LINK](https://AndriiYelieva.github.io/react_clock/)

### CODE REVIEW

**Review by VitaliyBodnar — CHANGES_REQUESTED** [review]
- the tests are not passed
Screenshot 2023-07-23 at 13 01 59

File: src/App.tsx [hunk 1]
```diff
@@ -1,39 +1,61 @@
...
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentWillUnmount(): void {
+    window.clearInterval(this.clockTimerId);
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    document.removeEventListener('contextmenu', this.hideClock);
```
Comment (IvanFesenko) [inline src/App.tsx:36]: Do not forget remove all added listeners - your forgot remove 'click'

File: src/components/Clock.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    const { clockName } = this.props;
+
+    return (
+      <div className="Clock">
+        <strong className="Clock__name">
+          {clockName}
+        </strong>
+
+        {' time is '}
+
```
Comment (IvanFesenko) [inline src/components/Clock.tsx:48]:
```suggestion
        <span> time is </span>
```

**Review by IvanFesenko — CHANGES_REQUESTED** [review]
GJ. Check my comments

**Review by mykhalenych — APPROVED** [review]

**Pull request closed without merging**